packages/client/lib/execution/vmexecution.ts

```diff
@@ -115,6 +115,7 @@ export class VMExecution extends Execution {
    * @param blocks Array of blocks to save pending receipts and set the last block as the head
    */
   async setHead(blocks: Block[]): Promise<void> {
+    await this.chain.blockchain.setIteratorHead('vm', blocks[blocks.length - 1].hash())
```

Comment:
Does moving this thing a fine lines before change anything?

Reply:
Yeah this is so when beaconsync gets CHAIN_UPDATED event from the chain.putBlocks, the vm execution will see the head matches the chain head and return with no further execution.

This is safe because the `runWithoutSetHead` already inserts the block in the blockchain db, so this method can safely reference that block as the new iterator head.

Reply:
Yup that makes sense!